ansible_base/authentication/backend.py

```diff
@@ -8,38 +9,36 @@
 
 logger = logging.getLogger('ansible_base.authentication.backend')
 
-authentication_backends = OrderedDict()
+
+@lru_cache(maxsize=1)
+def get_authentication_backends(last_updated):
+    # last_updated is primarily here as a cache busting mechanism
+    authentication_backends = OrderedDict()
+
+    for database_authenticator in Authenticator.objects.filter(enabled=True):
+        try:
+            authentication_backends[database_authenticator.id] = get_authenticator_plugin(database_authenticator.type)
+        except ImportError:
+            continue
+        authenticator_object = authentication_backends[database_authenticator.id]
+        authenticator_object.update_if_needed(database_authenticator)
+    return authentication_backends
 
 
 class AnsibleBaseAuth(ModelBackend):
     def authenticate(self, request, *args, **kwargs):
         logger.debug("Starting AnsibleBaseAuth authentication")
+        last_modified = Authenticator.objects.values("modified_on").order_by("-modified_on").first()["modified_on"]
 
-        for database_authenticator in Authenticator.objects.filter(enabled=True):
-            # Either get the existing object out of the backends or get a new one for us
-            if database_authenticator.id not in authentication_backends:
-                try:
-                    authentication_backends[database_authenticator.id] = get_authenticator_plugin(database_authenticator.type)
-                except ImportError:
-                    continue
-            else:
-                if authentication_backends[database_authenticator.id].type != database_authenticator.type:
-                    try:
-                        authentication_backends[database_authenticator.id] = get_authenticator_plugin(database_authenticator.type)
-                    except ImportError:
-                        # Since we failed to load this authenticator_plugin lets take it out of the cache
-                        del authentication_backends[database_authenticator.id]
-                        continue
-            authenticator_object = authentication_backends[database_authenticator.id]
-            authenticator_object.update_if_needed(database_authenticator)
+        for k, authenticator_object in get_authentication_backends(last_modified).items():
```

Comment:
Can we change `k` to something mroe descriptive like `authenticator_name` or something along those lines?